### PR TITLE
DAGView dynamic node scaling + secondary relationships

### PR DESCRIPTION
**Pull request by tharmaman**
- Add ability to specify primary and secondary relationships to a CompositeRelationship
- Make the graphBuilder work with secondary relationships
- Make RelationshipGridPlacementStrategy be able to take in an embeddedSecondaryRelationshipStrategy to render the secondary relationships
- Add a ScaleNodeSecondaryRelationshipStrategy.js which is the base for embeddedSecondaryRelationshipStrategy which will scale the node according to the length of the array returned after grabbing the secondary relationships
- Optionally render the arrowhead
- Allow the passing of 2 dimensions of a node size instead of just 1 dimension which would've allowed for only squares to be used
- Add the 2 multiplier back in the denominator for the cellLaneRatio

### CODE REVIEW

**Review by jlhughes — COMMENTED** [review]
This was a lot of work. Looks good.

**Review by sarthak-marwaha — APPROVED** [review]
LGTM

File: src/foam/graph/GraphBuilder.js [hunk 1]
```diff
@@ -52,13 +52,45 @@ foam.CLASS({
         if ( ! noRootAdd ) this.roots.push(this.data[rootObject.id]);
       }
 
+      var isRoot;
+      this.roots.forEach(root => {
+        if ( root.id === rootObject.id ) isRoot = true;
+      })
```
Comment (lchanmann) [inline src/foam/graph/GraphBuilder.js:58]: can use Array.prototype.some?

File: src/foam/graph/GraphBuilder.js [hunk 2]
```diff
@@ -52,13 +52,45 @@ foam.CLASS({
...
+
+          // Add secondary relationship link but don't create actual nodes for them
+          if ( compositeRelationship && compositeRelationship.getSecondaryForwardNames().length > 0 ){
+            // TODO: make this work with an array
+            var secondaryRelationshipKey = compositeRelationship.getSecondaryForwardNames()[0];
+
+            var secondaryRelationshipDAO = o[secondaryRelationshipKey].dao || o[secondaryRelationshipKey];
+            secondaryRelationshipDAO.select().then(secondaries => {
+              this.data[o.id].secondaryForwardLinks = secondaries.array ? secondaries.array : [secondaries];
+            })
```
Comment (lchanmann) [inline src/foam/graph/GraphBuilder.js:91]: Similar code to line 66-72, maybe extract a method?

File: src/foam/graph/map2d/ScaleNodeSecondaryRelationshipStrategy.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        }
+
+        return maybeAddMore;
+      }
+    },
+    {
+      name: 'getBaseCellSize',
+      code: function getBaseCellSize(obj) {
+        if ( obj.secondaryForwardLinks === undefined ) {
+          console.warn("obj.secondaryForwardLinks.length is undefined");
```
Comment (lchanmann) [inline src/foam/graph/map2d/ScaleNodeSecondaryRelationshipStrategy.js:72]: is it checking undefined for secondaryForwardLinks or secondaryForwardLinks.length?

**Pull request merged**